simple/bug_world/world.py

```diff
@@ -62,7 +63,11 @@ def step(self):
     # Bugs can eat or move in one step.
     # we need to know how many bugs are in one food grid. There's
     # a certain rule that how food are distributed among bugs.
-    for _, bug in self._bugs.items():
+    _current_bugs = copy.deepcopy(self._bugs)
```

Comment:
you can do this to avoid mutating the dictionary in the loop:
keys = tuple(self._bugs.keys())
then use keys in the loop.